File: Hayaa.ServiceFrame/ServiceFrameClient/Hayaa.ServicePlatform.Client/Util/Heart.cs
using Hayaa.BaseModel.Model;
using Hayaa.Common;
using Hayaa.ConfigSeed.Standard;
using Hayaa.Security.Client;
using Hayaa.ServicePlatform.Client.Config;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;

namespace Hayaa.ServicePlatform.Client.Util
{
    internal class Heart
    {
        private static Heart _isntance = new Heart();

        private Heart()
        {

        }
        internal static Heart Isntance { get => _isntance; }
        private static Timer g_timer = null;
        private void HeartAction(object state)
        {
            String response = null;
            var urlParamater = new Dictionary<string, string>();
            urlParamater.Add("t", g_token);
            urlParamater.Add("appinstanceid", g_appInstanceId.ToString());
            urlParamater.Add("appid", g_appId.ToString());
            response = HttpHelper.Transaction(ConfigHelper.Instance.GetComponentConfig().HeartServiceUrl, urlParamater);
            try
            {
                TransactionResult<List<int>> serviceResult = JsonHelper.DeserializeSafe<TransactionResult<List<int>>>(response);
                if (serviceResult.Code == 0)
                {

                    if (serviceResult.Data.Contains(1))
                    {
                        AppSeed.Instance.Resetonfig();
                    }
                    if (serviceResult.Data.Contains(2))
                    {
                        SecurityRoot.Reset(g_appId);
                    }
                }
            }
            catch
            {

            }
        }
        private String g_token;
        private int g_appInstanceId;
        private int g_appId;
        public void Start(String token, int appInstanceId, int appId)
        {
            g_token = token;
            g_appInstanceId = appInstanceId;
            g_appId = appId;
            g_timer = new Timer(HeartAction, null, ConfigHelper.Instance.GetComponentConfig().HeartTimespan * 1000, 5 * 60 * 1000);
        }
    }
}

File: Hayaa.ServiceFrame/ServiceFrameClient/hayaa.serviceplatform.client_java/hayaa.serviceplatform.client/src/main/java/hayaa/serviceplatform/client/Heart.java
package hayaa.serviceplatform.client;

import Hayaa.ConfigSeed.AppSeed;
import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.TypeReference;
import hayaa.basemodel.model.TransactionResult;
import hayaa.common.HttpHelper;
import hayaa.common.JsonHelper;

import java.util.*;

class Heart {
    public static Heart getInstance() {
        return instance;
    }

    private static Heart instance = new Heart();
    private static Timer g_timer = null;
    private String g_token;
    private Integer g_appInstanceId;
    private Integer g_appId;

    private void HeartAction() {
        String response = null;
        Map<String, String> urlParamater = new HashMap<String, String>();
        urlParamater.put("t", g_token);
        urlParamater.put("appinstanceid", g_appInstanceId.toString());
        urlParamater.put("appid", g_appId.toString());
        response = HttpHelper.Transaction(ConfigHelper.instance().GetComponentConfig().getHeartServiceUrl(), urlParamater, "post");
        try {
            TransactionResult<List<Integer>> serviceResult =(TransactionResult<List<Integer>>)JSON.parseObject(response, new TypeReference<TransactionResult<List<Integer>>>() {
            });
            if (serviceResult.getCode() == 0) {

                if (serviceResult.getData().contains(1)) {
                    AppSeed.Instance().Resetonfig();
                }
                if (serviceResult.getData().contains(2)) {
                    // SecurityRoot.Reset(g_appId);
                }
            }
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }
    public void Start(String token, int appInstanceId, int appId)
    {
        g_token = token;
        g_appInstanceId = appInstanceId;
        g_appId = appId;
        g_timer = new Timer();
        g_timer.schedule(new TimerTask() {
            @Override
            public void run() {
                HeartAction();
            }
        },60 * 1000,ConfigHelper.instance().GetComponentConfig().getHeartTimespan() * 1000*5  );
    }
}
